### ISSUE TITLE
Structure "planting"

### ISSUE THREAD

**Issue by phate09** [phate09/ToDoChores#10]
Some structures are better if correctly spaced and ordered
I suggest an action for building structures in the same way we implement planting
The structure to "plant" could be(do not need all in the list if not deemed necessary):
-Chests (!)
-Drying racks
-Pig houses
-Bunny houses
-Other suggestions?

Ofc we cannot check if the user has all the needed materials so it could be a chain of ifs for the recipes:
if wood->boards
if boards->chest

What do you think?

**Comment by taichunmin** [comment]
I can't figure out how to better then "Egg Pain Together" XD
I really admire that mod.

**Comment by taichunmin** [comment]
I buyed a lot game in Steam Summer Sale. So I will temporary play other games. XD

**Comment by phate09** [comment]
true

**Comment by 1to3fall5** [comment]
Hi guys, first of all great job there adding new feature, the mod is getting better and better. I was also playing around with the code the other day, i was think about asking u guys adding the plant structure feature too, but i remember said this mod is for personal use so i didn't ask phate  much about add other features. so its really not a problem to be better than the "Egg Pain Together" or not, that mod is not that perfect either. with that said there are some of my thoughts and ideas.

--(Gap adjustment and snap to grid) the mod EPT does't have much config for the gap between the planting, and the green shadow placer is not always correct, if there is a way to snap to the geometric placement mod grid it wound be great, while in TDC we have the code for changing the gap but it's not yet can be used for config, so i think that would be a better feature than EPT.

--(UI config) the one part  I like the EPT more, it's that the screen cleaner, i mean for now TDC has all the icon floating on the screen, that's the good and also the part that i don't like, i always change the UI to 0.8 scale after every update, as for EPT there is no such thing, but if we can change to UI so that we can dismiss the icon after we set the action it would be awesome.

--(other function, row 9,10...) for now the one feature i can't get from TDC is the pitch fork part, in EPT i can shift select the turf and then it will auto digging out or place the turf. also, the structure placement part, i change the code in TDC the other day and found out the current placer is not snap to the wall grid, i think in game there is a 0.5 difference. I made it work only when I set the gap bigger lol, EGT can place the wall nicely but the placer is not shown correctly. 

--(ideas) the first idea i had after i used TDC is that i think this might be a good game mode for DST, let me explain. so now we can use the button to control character to proceed the actions, to let them survive other than we decide where we go. The only thing we have to do is to let them get the right action in the right time, sounds like the other survival game right, "oxygen not included", actually a i think that's how they got the idea for that game, while they debugging the game the figured out the concept haha.

ya, those are some of my thoughts, i really hope I can contribute to this mod, but for now i can only give some ideas, sorry if there is any English mistakes, oh also, if u can change to name of the mod to like "auto action" or something it might be more catchy.

**Comment by phate09** [comment]
> --(UI config) the one part I like the EPT more, it's that the screen cleaner, i mean for now TDC has all the icon floating on the screen, that's the good and also the part that i don't like, i always change the UI to 0.8 scale after every update, as for EPT there is no such thing, but if we can change to UI so that we can dismiss the icon after we set the action it would be awesome.

Yeah well i had in mind the idea of checkboxes and then just press another shortcut to "commit" the actions desired. In this way we would not need to keep the widget icons open the whole time
What do you think?

**Comment by 1to3fall5** [comment]
Maybe that would work but I'm not quite sure how it looks, it's hard to imagine that through words. oh an idea, maybe make it a bit like the current in game craft side bar ? you know like it will have different tab and if you click it, it will expend to show the options, but that might be a lot of work for the coding part i suppose.
